### PR TITLE
Feature/Infinite-scrolling-feature-and-user-search-fix-on-user-management-page

### PR DESCRIPTION
**Pull request by prashant67690**
Date: 28-09-2024

Developer Name: Prashant Kumar 

---

## Issue Ticket Number

closes : [#833](https://github.com/Real-Dev-Squad/website-dashboard/issues/833)  
## Description
The User Management page currently features infinite scroll loading of users and the users in the list view are now rendered as the list rather than in the form of cards / tiles , due these changes the search functionality was having issue , that thing is fixed and running as expected down below.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

Before:
prev
oldlist
371653915-742a5a61-42d0-48d9-a1a1-058c21169ad7

</details>
<details>
<summary>Screenshot 2</summary>

After:
loader
afterlist

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

tests
</details>

## Additional Notes

here we can see the search functionality behavior after the fixes 

https://github.com/user-attachments/assets/847504bb-caa7-4589-8f03-d7cd69437f03

### CODE REVIEW

**Review by vinit717 — CHANGES_REQUESTED** [review]

File: users/script.js
[the remark this answers is not in the document]
Comment (prashant67690) [reply users/script.js:1]: which functionality , for checking weather user loader is happening or not is done , i just fixed some other thing that broke due to my change in the logic of loading more users

File: users/script.js [hunk 1]
```diff
@@ -162,10 +160,18 @@ function generateUserList(
     return;
   }
 
-  const ulElement = document.getElementById('head_list');
+  console.log(users);
```
[the remark this answers is not in the document]
Comment (prashant67690) [reply users/script.js]: fixed this

File: users/script.js [hunk 2]
```diff
@@ -162,10 +160,17 @@ function generateUserList(
     return;
   }
 
-  const ulElement = document.getElementById('head_list');
+  const errorTexts = document.getElementsByClassName('error-text');
+  Array.from(errorTexts).forEach((errorText) => {
+    errorText.remove();
+  });
```
Comment (yesyash) [inline users/script.js:166]: - do not select the element using classname use id instead. 
- as we know there will be only one element that will contain the error message do we need to create an array and loop over all elements that match the classname?

**Review by yesyash — CHANGES_REQUESTED** [review]

File: users/script.js [hunk 3]
```diff
@@ -162,10 +160,17 @@ function generateUserList(
...
+
+  if (showPagination || page == 0) {
+    ulElement.innerHTML = '';
+  }
 
-  if (users != null) {
-    users.forEach((userData) => {
+  if (users?.length !== 0) {
```
Comment (yesyash) [inline users/script.js:172]: - do we need an explicit `!== 0` check here?

File: users/script.js [hunk 4]
```diff
@@ -31,21 +32,24 @@ const init = (
...
         return;
       }
-      run = false;
-      showUserDataList(
-        page++,
-        userListElement,
-        paginationElement,
-        loaderElement,
-      );
+      if (page === 0) {
```
Comment (yesyash) [inline users/script.js:42]: - do we need an explicit `===0` check here?

File: users/script.js [hunk 5]
```diff
@@ -31,21 +32,24 @@ const init = (
...
+        page++;
```
Comment (yesyash) [inline users/script.js:43]: - can we do an early return here?

**Review by yesyash — CHANGES_REQUESTED** [review]

**Comment by prashant67690** [comment]
yash sir fixed the previous problems and also wrote the additional test as you requested please review

File: users/style.css [hunk 6]
```diff
@@ -145,10 +149,21 @@ li p {
...
+  height: 50px;
+  animation: spin 2s linear infinite;
+  margin: 150px auto;
+}
+
 .pagination {
   display: flex;
   justify-content: space-between;
   margin-top: auto;
+  visibility: hidden;
```
Comment (Ayushsanjdev) [inline users/style.css:166]: You know that this element will still take the space if we use visibility hidden? Is it the use case? can you explain? Why not display: none

**Review by Ayushsanjdev — CHANGES_REQUESTED** [review]

File: users/style.css (hunk 6 above)
Comment (prashant67690) [reply users/style.css:166]: did this so other features that might depend on it are not broken and i can develop the required feature and there were some services that depend on it

**Review by Ayushsanjdev — APPROVED** [review]
LGTM

**Review by yesyash — APPROVED** [review]

**Review by vinit717 — APPROVED** [review]

**Pull request merged**